packages/table/src/common/utils.ts

```diff
@@ -259,6 +259,32 @@ export const Utils = {
     },
 
     /**
+     * Returns a descriptive object for each key whose values are shallowly unequal between two
+     * provided objects. Useful for debugging shouldComponentUpdate.
```

Comment:
I'm fine with adding some handy utility methods for debugging to the codebase. It does add some bytes to the download, but so long as we don't go crazy it wont be a big deal.

Do you think we should unit test these things? (that does coverage say?)